feat(kmp): improve spm docs (#9360)
getsentry/sentry-docs: docs/platforms/kotlin-multiplatform/troubleshooting.mdx

---
title: Troubleshooting
description: "Troubleshoot and resolve edge cases."
sidebar_order: 9000
---

> If you need help solving issues with your Sentry Kotlin Multiplatform SDK, you can read the edge cases documented below. If you need additional help, you can [ask on GitHub](https://github.com/getsentry/sentry-kotlin-multiplatform/issues/new/choose). Customers on a paid plan may also contact support.

Reviewer: ```suggestion
This document covers common issues you may encounter when using the Sentry Kotlin Multiplatform SDK and provides steps to troubleshoot.

If you need additional help, you can [ask on GitHub](https://github.com/getsentry/sentry-kotlin-multiplatform/issues/new/choose). Customers on a paid plan may also contact support.
```
I feel like edge cases aren't really covered here

Reviewer 2: Yeah, I think since for now we're just listing one specific problem, we should just talk about it directly as opposed to calling it an edge case. :)

## Tests not working

If you configured the Sentry Kotlin Multiplatform SDK and tests are still failing with the following error: `ld: framework 'Sentry' not found` then follow these steps for the workaround:

### 1. Download the Sentry xcframework

> Choose the release based on the version specified in the version compatibility table, download the `Sentry.xcframework.zip`, and unzip it.

Reviewer: ```suggestion
Choose the release based on the version specified in the [version compatibility table](<link>), download the `Sentry.xcframework.zip`, and unzip it.
```
can you add a link to the table?

> ### 1. Download the Sentry xcframework
> 
> Choose the release based on the version specified in the version compatibility table, download the `Sentry.xcframework.zip`, and unzip it.

Reviewer: ```suggestion
1. Ensure you have a compatible Sentry xcframework. Choose the release based on the version specified in the version compatibility table, download the `Sentry.xcframework.zip`, and unzip it.
```
I would consider reformatting the headings into ordered list items, since the instructions are each one step and aren't long enough to benefit much from clickable headings. If we add more troubleshooting items later, I think it'll also look cleaner.

### 2. Create Frameworks directory and insert Sentry.framework

Create a `/Frameworks` directory in the directory where the `test.kexe` resides. Put the `Sentry.framework` into it. (The `Sentry.framework` can be found inside of the `ios-arm64_x86_64-simulator`.)
The `test.kexe` will usually reside in `build/bin/iosSimulatorArm64/debugTest`.

> ### 3. Add the linker options

Reviewer: ```suggestion
### 3. Add linker options
```

> 

Reviewer: ```suggestion

Modify `shared/build.gradle.kts` to include linker options, as shown below:

```

---
title: Troubleshooting

> description: "Troubleshoot and resolve edge cases."

Reviewer 2: ```suggestion
description: "Troubleshoot and resolve edge cases when using Sentry's Kotlin Multiplatform SDK."
```

sidebar_order: 9000
---

This document covers common issues you may encounter when using the Sentry Kotlin Multiplatform SDK and provides steps to troubleshoot.

> If you need additional help, you can [ask on GitHub](https://github.com/getsentry/sentry-kotlin-multiplatform/issues/new/choose). Customers on a paid plan may also contact support.
> 

Reviewer 2: ```suggestion
If you need additional help, [ask on GitHub](https://github.com/getsentry/sentry-kotlin-multiplatform/issues/new/choose). Customers on a paid plan can also contact [support](https://help.sentry.io/).

```

> ## Tests not working

Reviewer 2: ```suggestion
## Tests Not Working
```

If you configured the Sentry Kotlin Multiplatform SDK and tests are still failing with the following error: `ld: framework 'Sentry' not found` then follow these steps for the workaround:

<Note>
This example shows you how to fix the issue for the iOS simulator target, but the same approach can be used for other targets.
</Note>

> 1. Download the Sentry xcframework

Reviewer 2: ```suggestion
**1. Download the Sentry xcframework**
```

> Ensure you have a compatible Sentry xcframework. Choose the release based on the version specified in the [version compatibility table](/platforms/kotlin-multiplatform/#cocoa-sdk-version-compatibility-table), download the `Sentry.xcframework.zip`, and unzip it.

Reviewer: Add a tab to nest this paragraph under the step above (same for the additional details under the steps below)

> 2. Create Frameworks directory and insert Sentry.framework

Reviewer 2: ```suggestion
**2. Create a frameworks directory and insert `Sentry.framework`**
```

> Create a `/Frameworks` directory in the directory where the `test.kexe` resides. Put the `Sentry.framework` into it. (The `Sentry.framework` can be found inside of the `ios-arm64_x86_64-simulator`.)
> The `test.kexe` will usually reside in `build/bin/iosSimulatorArm64/debugTest`.

Reviewer 2: ```suggestion
Create a `/Frameworks` directory in the directory where the `test.kexe` resides and put the `Sentry.framework` in it. 
- The `Sentry.framework` can be found inside of the `ios-arm64_x86_64-simulator`.
- The `test.kexe` will usually reside in `build/bin/iosSimulatorArm64/debugTest`.
```

> 3. Add linker options

Reviewer 2: ```suggestion
**3. Add linker options**
```

Modify `shared/build.gradle.kts` to include linker options, as shown below:

```kotlin {filename:shared/build.gradle.kts}
  listOf(
      iosX64(),
      iosArm64(),
      iosSimulatorArm64(),
  ).forEach {
      it.binaries.framework {
          baseName = "shared"
      }
      it.compilations.all {
          if (compilationName == "test" && target.platformType == KotlinPlatformType.native) {
              compilerOptions.configure {
                  freeCompilerArgs.add("-linker-options")
                  freeCompilerArgs.add("-F/your/path/Carthage/Build/Sentry.xcframework/ios-arm64_x86_64-simulator/")
              }
          }
      }
  }
```

> 4. Run the tests

Reviewer 2: ```suggestion
**4. Run the tests**
```
